[NET-5325] ACL templated policies support in tokens and roles (#18708)
hashicorp/consul: .changelog/18708.txt

```release-note:feature
acl: Added ACL Templated policies to simplify getting the right ACL token.
```

```release-note:improvement

> cli: A `-templated-policy`, `-templated-policy-file`, `-replace-templated-policy`, `-append-templated-policy`, `-replace-templated-policy-file`, `-append-templated-policy-file` and `-var` flags can now be used when creating or updating tokens/roles.

Reviewer: ```suggestion
cli: Added `-templated-policy`, `-templated-policy-file`, `-replace-templated-policy`, `-append-templated-policy`, `-replace-templated-policy-file`, `-append-templated-policy-file` and `-var` flags for creating or updating tokens/roles.
```
